deps/rabbitmq_mqtt/test/auth_SUITE.erl

```diff
@@ -588,7 +591,7 @@ client_id_from_cert_dn(Config) ->
 invalid_client_id_from_cert_san_dns(Config) ->
     MqttClientId = <<"other_client_id">>,
     {ok, C} = connect_ssl(MqttClientId, Config),
-    ?assertMatch({error, _}, emqtt:connect(C)),
+    {error, {client_identifier_not_valid, _}} = emqtt:connect(C),    
```

Comment:
This PR doesn't fix the flake. I can repro the flake locally as follows:
```erl
    {error, {client_identifier_not_valid, _}} = emqtt:connect(C),
    timer:sleep(1),
    unlink(C).
```
which makes the test fail with:
```
=== === Reason: {'EXIT',{shutdown,client_identifier_not_valid}}
```
That's the flake which we also observed in CI.

The correct fix is to unlink first, then connect.

Reply:
i cannot reproduce it locally and i dont see CI failing because of this . I can modify the test as you suggest though.

Reply:
Can you send me the link to the failed job in CI where this test case is failing ?

Reply:
> Can you send me the link to the failed job in CI where this test case is failing ?

Yes, here are two flakes failing due to `{'EXIT',{shutdown,client_identifier_not_valid}}`:
* https://github.com/rabbitmq/rabbitmq-server/actions/runs/12769170662/job/35591148832
* https://github.com/rabbitmq-pro/tanzu-rabbitmq/actions/runs/12768915497/artifacts/2428404146

Reply:
> i cannot reproduce it locally and i dont see CI failing because of this .

The CT test case process runs concurrently with the MQTT connection process. CI can fail if the CT test case process executes `unlink(C)` after receiving the EXIT signal.

Reply:
> I still dont understand what you are suggesting. 

I'm suggesting to flip the two lines, i.e. to unlink **before** the expected failure as done in many other places in this test suite, e.g. in:
* https://github.com/rabbitmq/rabbitmq-server/blob/3a61274f0311bd4ff26fbc98e4e5cc9b855c9837/deps/rabbitmq_mqtt/test/auth_SUITE.erl#L810-L812
* https://github.com/rabbitmq/rabbitmq-server/blob/3a61274f0311bd4ff26fbc98e4e5cc9b855c9837/deps/rabbitmq_mqtt/test/auth_SUITE.erl#L945-L947
* https://github.com/rabbitmq/rabbitmq-server/blob/3a61274f0311bd4ff26fbc98e4e5cc9b855c9837/deps/rabbitmq_mqtt/test/auth_SUITE.erl#L1227-L1228

Reply:
those ci failures are because this change has not been merged yet

Reply:
Regardless i have committed that change.